src/handlers.rs

```diff
@@ -393,6 +397,138 @@ where
+            let (mut original_psbt, _tx_details) = tx_builder
+                .finish()
+                .map_err(|e| Error::Generic(format!("Unable to build transaction: {}", e)))?;
+
+            // check + sign psbt
+            let finalized = wallet.sign(&mut original_psbt, SignOptions::default())?;
+
+            // handle error if the psbt is not finalized
+            if !finalized {
+                return Err(Error::Generic("PSBT not finalized".to_string()));
+            }
+
+            // set payjoin params
+            // TODO use fee_rate
+            let pj_params = payjoin::send::Configuration::with_fee_contribution(
+                payjoin::bitcoin::Amount::from_sat(100),
+                None,
+            );
+
+            // save a clone for after processing the response
+            let mut ocean_psbt = original_psbt.clone();
+
+            // Construct the request with the PSBT and parameters
+            let (req, ctx) = uri
+                .create_pj_request(original_psbt, pj_params) // not clone here
+                .map_err(|e| Error::Generic(format!("Error building payjoin request: {}", e)))?;
+
+            // reqwest client
+            let client = reqwest::blocking::Client::builder()
+                .danger_accept_invalid_certs(true)
+                .build()
+                .map_err(|e| Error::Generic(format!("Error building payjoin client: {}", e)))?;
+
+            // send the request, and receive response
+            let response = client
+                .post(req.url)
+                .body(req.body)
+                .header("Content-Type", "text/plain")
+                .send()
+                .map_err(|e| Error::Generic(format!("Error with HTTP request: {}", e)))?;
+
+            // process the response
+            let mut payjoin_psbt = ctx
+                .process_response(response)
+                .map_err(|e| Error::Generic(format!("Error processing payjoin response: {}", e)))?;
+
+            // need to reintroduce utxo from original psbt
+            fn input_pairs(
+                psbt: &mut PartiallySignedTransaction,
+            ) -> Box<dyn Iterator<Item = (&bdk::bitcoin::TxIn, &mut Input)> + '_> {
+                Box::new(psbt.unsigned_tx.input.iter().zip(&mut psbt.inputs))
+            }
+
+            // get original inputs from original psbt clone (ocean_psbt)
+            let mut original_inputs = input_pairs(&mut ocean_psbt).peekable();
+
+            for (proposed_txin, proposed_psbtin) in input_pairs(&mut payjoin_psbt) {
+                if let Some((original_txin, original_psbtin)) = original_inputs.peek() {
+                    if proposed_txin.previous_output == original_txin.previous_output {
+                        proposed_psbtin.witness_utxo = original_psbtin.witness_utxo.clone();
+                        proposed_psbtin.non_witness_utxo = original_psbtin.non_witness_utxo.clone();
+                    }
+                    original_inputs.next();
+                }
```

Comment:
```suggestion
                        proposed_psbtin.non_witness_utxo = original_psbtin.non_witness_utxo.clone();
                         original_inputs.next();
                    }
                }
```

Oops! This will skip checking an original_input even if it's not a match, then failing will sign if a match is missed.